kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotManager.java

```diff
@@ -642,19 +648,24 @@ protected Optional<LogSegment> getLogSegmentForVersion(
 
         // We may just be getting a checkpoint file after the filtering
         if (!deltaVersionsAfterCheckpoint.isEmpty()) {
+            // If we have deltas after the checkpoint, the first file should be 1 greater than our
+            // last checkpoint version. If no checkpoint is present, this means the first delta file
+            // should be version 0.
             if (deltaVersionsAfterCheckpoint.getFirst() != newCheckpointVersion + 1) {
-                throw new RuntimeException(
-                    format(
-                        "Log file not found.\nExpected: %s\nFound: %s",
-                        FileNames.deltaFile(logPath, newCheckpointVersion + 1),
-                        FileNames.deltaFile(logPath, deltaVersionsAfterCheckpoint.get(0))
+                throw new InvalidTableException(
```

Comment:
`DeltaErrors.invalidTable(Path tablePath, String msg, Object args...)`?

Reply:
I wasn't sure if we needed a method for this. If it just passes the args exactly the same and it's a specific exception class it seems like we can just throw the exception in line? What do you think?

Reply:
sounds good.